src/lib.rs

```diff
@@ -1790,31 +1795,36 @@ pub mod __rt {
     ///
     /// Ideas for how to improve this are most welcome!
     pub fn link_mem_intrinsics() {
-        crate::externref::link_intrinsics();
-    }
-
-    std::thread_local! {
-        static GLOBAL_EXNDATA: Cell<[u32; 2]> = Cell::new([0; 2]);
+        #[cfg(feature = "std")]
+        {
+            crate::externref::link_intrinsics();
+        }
```

Comment:
I'm incredibly mindblown by this hack and very surprised its needed in the first place.
The [`no_mangle` attribute](https://doc.rust-lang.org/1.80.0/reference/abi.html?highlight=no_mangle#the-no_mangle-attribute) should imply the [`used` attribute](https://doc.rust-lang.org/1.80.0/reference/abi.html?highlight=no_mangle#the-used-attribute), addressing this issue.

Without digging into this, the best we can do is make a separate module, as described in the commend, and move `link_intrinsics()` there.